glide-core/src/client/mod.rs

```diff
@@ -980,6 +980,56 @@ async fn create_cluster_client(
         }
     }
 
+    // Handle database selection for cluster mode
+    // If database_id is not 0 (default), we need to select the database on all nodes
+    if redis_connection_info.db != 0 {
```

Comment:
If the client is lazy, should this actually happen now or should it be deferred?

Reply:
This piece of the code was unnecessary and it was removed. 
The select happen in the connection.rs, which was already there, I had to remove one hardcoded configuration for the cluster specifically that was always setting to 0 for the reconnection.
My understanding is that the connection is deferred for the lazy connect, but I didn't test this case specifically